docs/test/writing.md

```diff
@@ -426,6 +426,54 @@ test("exactly two assertions", () => {
 
 This helps ensure all your assertions run, especially in complex async code with multiple code paths.
 
+## Type Testing
+
+Bun includes `expectTypeOf` for testing typescript types, compatible with Vitest.
+
+### expectTypeOf
+
+{% callout %}
+
+**Note** — These functions are no-ops at runtime - you need to run TypeScript separately to verify the type checks.
+
+{% endcallout %}
+
+The `expectTypeOf` function provides type-level assertions that are checked by TypeScript's type checker. **Important**:
+
+To test your types:
+
+1. Write your type assertions using `expectTypeOf`
+2. Run `bunx tsc --noEmit` to check that your types are correct
```

Comment:
This command involves two runtimes and one compiler. Three great pieces of software, written by people much more experience than me. Unfortunately none of them is a type test runner, so having a simple `test.only`, `test.skip` will not work.

In case you are curious, a type test runner can be found here: https://tstyche.org. Use `test.only`, run a single file, test agains specific TypeScript version, generate test files using a data table, etc. You found it! (;